src/commands/howto.ts

```diff
@@ -0,0 +1,151 @@
+  "howto",
+  commands.map((c) => c.name)
+);
+
+export const data = async () => {
+  const addTargetUserParam = (scb: SlashCommandSubcommandBuilder) =>
+    scb.addUserOption((b) =>
+      b.setName("user").setDescription("User to send the message to").setRequired(false)
+    );
+
+  let rootCommandBuilder = new SlashCommandBuilder()
+    .setName("howto")
+    .setDescription("HOWTOs and tutorials");
+
+  for (let subcommand of commands) {
+    rootCommandBuilder.addSubcommand((scb) =>
+      addTargetUserParam(scb.setName(subcommand.name).setDescription(subcommand.description))
+    );
+  }
+  return rootCommandBuilder.toJSON();
+};
+
+const allowedRoles = new Set(["admin", "mods", "mods+", "power-users"]);
+
+const hasSufficientPrivilege = (member: GuildMember | APIInteractionGuildMember | null) => {
+  let guildMember: GuildMember = member as GuildMember;
+  return guildMember && guildMember.roles.cache.some((role) => allowedRoles.has(role.name));
+};
+
+export const call = async (interaction: ChatInputCommandInteraction) => {
+  let interactionReply = async (content: string, ephemeral: boolean = true) => {
+    await interaction.reply({ content, ephemeral });
+  };
+
+  let invokingUser = interaction.user;
+  let targetUser = interaction.options.getUser("user", false) ?? invokingUser;
+
+  if (targetUser.bot) {
+    await interactionReply(
+      `${userMention(invokingUser.id)}, you cannot use this command on a bot.`
+    );
+    return;
+  }
+
+  let selfTarget = targetUser.id === invokingUser.id;
+  if (selfTarget || hasSufficientPrivilege(interaction.member)) {
+    let subCommand = interaction.options.getSubcommand();
+    let dmReply = commands.find((c) => c.name == subCommand);
+
+    if (dmReply) {
+      try {
+        await postHowtoDm(dmReply, targetUser);
+        await interactionReply(`${userMention(targetUser.id)} please check your DMs`, selfTarget);
+      } catch (reason) {
+        let url = `https://github.com/codewars/discord-bot/blob/main/text/howto/${subCommand}.md`;
+        await interactionReply(
+          `${userMention(targetUser.id)} I couldn't DM you. See ${hyperlink(
+            dmReply.description,
+            hideLinkEmbed(url)
+          )} instead.`,
+          selfTarget
+        );
+      }
+    } else {
+      await interactionReply(`Unknown command: \`${subCommand}\``);
+    }
+  } else {
+    await interactionReply(
+      `${userMention(invokingUser.id)}, you are not privileged to use this command.`
+    );
+  }
+};
+
+const postHowtoDm = async (command: HowtoCommand, targetUser: User) => {
+  let body = howtoTexts.get(command.name);
+  if (!body) {
+    return;
+  }
+  let message = await targetUser.send(body);
+  for (let reaction of command.reactions) {
+    message.react(reaction.emoji);
+  }
+
+  // Do not set up a collector if there are no reactions
+  if (!command.reactions.length) return;
+
+  const collector = message.createReactionCollector({ filter: (_, reactor) => !reactor.bot });
```

Comment:
The "forever" was somewhat deliberate, because if the collector would be to expire, the _"react with `:emoji:` for more info"_ and reaction hints would become invalid. I can set up an expiration time for the collector, but would you have an idea what to do with the parts of the message which suggest follow-up reactions? Leave them as they are so they cease to work after expiration, or remove them by editing the message after that time, or maybe remove the follow-up reactions completely to get rid of the problem?
What expiration time would you suggest?

Reply:
I understand, but you can't expect it to work forever, so we need to clean up.

I did something similar for the trainer link detection. The bot adds reaction to the message for users to just click, waits for reaction, and cleans up to avoid any confusion in the future. See https://github.com/codewars/discord-bot/blob/main/src/events/messageCreate/handlers/trainer-link.ts

Reply:
Yeah i know that trainer detection works this way but tbh I personally dont like it because it causes text to move around in an annoying way. But sure, i can try and well see how it works.

Reply:
I walso wonder if theres anything special, wrt privileges or anything, about editing DMs rather than messages in a channel.

Will check.

Reply:
Maybe strike-through instead of removing? If you're talking about adding and removing the reaction, you don't need to do it.